server/gitrest/src/utils/helpers.ts

```diff
@@ -30,3 +31,54 @@ export function getExternalWriterParams(params: string | undefined): IExternalWr
     }
     return undefined;
 }
+
+export async function exists(fileSystemManager: IFileSystemManager, fileOrDirectoryPath: PathLike): Promise<boolean> {
+    try {
+        await fileSystemManager.stat(fileOrDirectoryPath);
+        return true;
+    } catch (error) {
+        if (error?.code === "ENOENT") {
+            // File/Directory does not exist.
+            return false;
+        }
+        throw error;
+    }
+}
+
+const latestFullSummaryFilename = "latestFullSummary";
+const getLatestFullSummaryFilePath = (dir: string) => `${dir}/${latestFullSummaryFilename}`;
+
+export async function persistLatestFullSummaryInStorage(
+    fileSystemManager: IFileSystemManager,
+    storageDirectoryPath: string,
+    latestFullSummary: IWholeFlatSummary,
+): Promise<void> {
+    const directoryExists = await exists(fileSystemManager, storageDirectoryPath);
```

Comment:
It might also be worth validating if besides existing, the `storageDirectoryPath` actually points to a directory:
```ts
if (!directoryExists.isDirectory()) {... }
```